core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java

```diff
@@ -57,20 +57,26 @@ private String generateFilename() {
         String.format("%05d-%d-%s-%05d", partitionId, taskId, uuid, fileCount.incrementAndGet()));
   }
 
+  /**
+   * Generates EncryptedOutputFile with relative path under iceberg table location.
+   */
+  public EncryptedOutputFile newOutputFile(String relativePath) {
```

Comment:
This class needs to create full paths. It should not create an output file for a relative path like this. That's why the location provider is used in the other cases.

Also, as I noted below, metadata files should not use the encryption manager and should also use `TableOperations.metadataFileLocation` to create a full path from a manifest file name. So I don't think we will need to add a method here.

That is, unless you wanted to rename this to `newMetadataOutputFile`, call `ops.metadataFileLocation` to create the full path, and bypass the encryption manager.